### DISCUSSION TITLE
Handling multiple compatibility terms

### DISCUSSION THREAD

**Discussion by Djiit** [renovatebot/renovate#31650] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Github, Renovate v38.97.1

### Please tell us more about your question or problem

Hello team,

I'm trying to instruct Renovate to update a Docker image based on 2 parts of the tags. The image is hosted on an ECR and Renovate is able to auth against it without any issue.

Here is a typical image: `ruby:3.2.2-bookworm-20240805_220331`

The first part of the tag is handled by the default dockerfile configuration: Renovate will look for 3.2.2+ tags, and leave the rest untouched. Great.

I'd like it to also manage "OS update", when we bump the image to a more recent version of Debian bookworm. I tried this packageRule without any luck: 
```
    {
      "matchManagers": ["dockerfile"],
      "matchPackageNames": [
        "<ecr-hostname>/ruby"
      ],
      "versionCompatibility": "^(?<compatibility>.*-.*)-(?<version>.*)$",
      "versioning": "loose",
    }
```

Is it possible to have 2 rules applied to the same dependency? and if yes, is compatibility the correct way to achieve that?

Thanks

**Comment by rarkins** [comment, marked as answer]
It's not possible, so you need to create a custom manager for updating the compatibility part

**Reply by Djiit** [reply]
Clear, thanks. Any pointer or example on top of https://docs.renovatebot.com/modules/manager/regex/ ?

**Reply by rarkins** [reply]
Not handy, but it's definitely been mentioned in past discussions in this repo previously at times

**Reply by Djiit** [reply]
Alright, I'll look for examples here. Thx
